ql/src/java/org/apache/hadoop/hive/ql/parse/ImportSemanticAnalyzer.java

```diff
@@ -1108,7 +1105,7 @@ private static void createRegularImportTasks(
           if (tblDesc.getLocation() != null) {
             tablePath = new Path(tblDesc.getLocation());
           } else {
-            tablePath = wh.getDefaultTablePath(parentDb, tblDesc.getTableName(), tblDesc.isExternal());
+            tablePath = getTableDataLocation(wh, tblDesc, x);
```

Comment:
why don't you move above `if (tblDesc.getLocation() != null)` to the helper method?